mrp_partial_production/models/mrp.py

```diff
@@ -62,60 +38,19 @@ def get_qty_available_to_produce(self):
         """Compute the total available to produce considering
         the lines reserved
         """
-        uom_obj = self.env['product.uom']
+        qty = []
         for record in self:
-            bom_obj = self.env['mrp.bom']
-            if not record.move_lines.mapped('reserved_quant_ids'):
-                return 0.0
-            self._cr.execute('''
-                             SELECT m.product_id,
-                                    sum(q.qty) AS total
-                             FROM stock_move AS m
-                             LEFT OUTER JOIN stock_quant AS q ON
-                                             q.reservation_id=m.id
-                             WHERE m.raw_material_production_id = {prod_id}
-                             GROUP BY m.product_id;
-                             '''.format(prod_id=record.id))
-            result = {i.get('product_id'): i.get('total') or 0
-                      for i in self._cr.dictfetchall()}
-            product_uom_qty = uom_obj._compute_qty(record.product_uom.id,
-                                                   1,
-                                                   record.product_id.uom_id.id)
-            consume_lines = bom_obj._bom_explode(record.bom_id,
-                                                 record.product_id,
-                                                 product_uom_qty)
-
-            qty = []
-            for line in consume_lines and consume_lines[0] or ():
-                product_id = line.get('product_id')
-                val = line.get('product_qty') and \
-                    int(result.get(product_id, 0) /
-                        line.get('product_qty')) or 0
+            result = {line.product_id.id: line.reserved_availability
+                      for line in record.move_raw_ids}
+            quantity = record.product_uom_id._compute_quantity(
+                record.product_qty, record.bom_id.product_uom_id)
+            boms, lines = record.bom_id.explode(record.product_id, quantity)
+
+            for line, line_data in lines:
+                product = line.product_id.id
+                original_qty = line.product_qty
+                val = original_qty and \
```

Comment:
1. to avoid `and` `or` you can use `if` `else`
2. to avoid `\` use `()`

Reply:
ready change